internal/templates/mismatchingselector/template.go

```diff
@@ -0,0 +1,55 @@
+package mismatchingselector
+
+import (
+	"fmt"
+
+	"golang.stackrox.io/kube-linter/internal/extract"
+
+	"golang.stackrox.io/kube-linter/internal/check"
+	"golang.stackrox.io/kube-linter/internal/diagnostic"
+	"golang.stackrox.io/kube-linter/internal/lintcontext"
+	"golang.stackrox.io/kube-linter/internal/objectkinds"
+	"golang.stackrox.io/kube-linter/internal/templates"
+	"golang.stackrox.io/kube-linter/internal/templates/mismatchingselector/internal/params"
+	metaV1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+)
+
+func init() {
+	templates.Register(check.Template{
+		HumanName:   "Mismatching Selector",
+		Key:         "mismatching-selector",
+		Description: "Flag deployments which do not have any selector matching a label in pod template",
+		SupportedObjectKinds: check.ObjectKindsDesc{
+			ObjectKinds: []string{objectkinds.DeploymentLike},
+		},
+		Parameters:             params.ParamDescs,
+		ParseAndValidateParams: params.ParseAndValidate,
+		Instantiate: params.WrapInstantiateFunc(func(_ params.Params) (check.Func, error) {
+			return func(_ *lintcontext.LintContext, object lintcontext.Object) []diagnostic.Diagnostic {
+				selector, hasSelector := extract.Selector(object.K8sObject)
+
+				if !hasSelector {
+					return []diagnostic.Diagnostic{{
+						Message: "Object has no selector specified",
+					}}
+				}
+				podTemplateSpec, hasPods := extract.PodTemplateSpec(object.K8sObject)
+				if !hasPods {
+					return nil
+				}
+				labelSelector, err := metaV1.LabelSelectorAsSelector(&metaV1.LabelSelector{MatchLabels: selector.MatchLabels, MatchExpressions: selector.MatchExpressions})
+				if err != nil {
+					return []diagnostic.Diagnostic{{
+						Message: fmt.Sprintf("Object has invalid label selector: %v", err),
```

Comment:
```suggestion
						Message: fmt.Sprintf("object has invalid label selector: %v", err),
```

We're trying to keep diagnostic messages starting with lower-case.